### ISSUE TITLE
Synchronising two different browsers to play audio at the same time

### ISSUE THREAD

**Issue by solarnz** [dhowden/tchaik#8]
- Is this something you would want in the application
- Could use webrtc for communication
- Other issues I don't know about

**Comment by dhowden** [comment]
Yes!  I had actually wanted to do something very similar, though maybe not quite as grand as using WebRTC - awesome idea!

My plan was roughly split into a few stages (based purely on my use cases):

1. REST-like API for remote-controlling the full player UI (done).
2. Remote control UI to control playback and show current/next track details (simple enough to be used on a phone).
3. "Headless" player which you could push the current track and play/pause/skip etc (i.e. running on a Raspberry Pi which would be plugged into an ADC+Amplifier), could be HTML5+Websocket setup or other process listening on a TCP socket... 
4. Several "headless" players on a network which would be synchronised (so that you could have the same music playing on different machines at the same time).

What you are asking for seems like a combination of **2**, **3** and **4**!

Based on what was implemented for **1** (backend sending control commands down the Websocket to the UI), it should be pretty straightforward to get **2** and **3** working for what you want.

**Comment by dhowden** [comment]
The last few commits have added much of this functionality (b8e8a67...e4436cc), though it isn't quite all hooked up yet!  Still need to add a front-end UI for setting up command "pushing" from the front-end.

There is currently a simple interface for setting up the front-end player key. I think it would be a good idea to make this into a more general pane including the ability to enter a player key which will be used to "publish" control events down the websocket.

It should also be possible to "un hook" the front-end player from the front-end controls (for the situation where you just want the remote player to be used).
